packages/dds/tree/src/shared-tree/treeApi.ts

```diff
@@ -67,7 +74,7 @@ export interface TreeApi extends TreeNodeApi {
  */
 export const treeApi: TreeApi = {
 	...treeNodeApi,
-	runTransaction<TNode extends TreeNode, TRoot>(
+	runTransaction<TNode extends TreeNode, TRoot extends ImplicitFieldSchema>(
```

Comment:
This method does not have a signature involving the root node type from the view, which was the problem that lead to the changes above. It resulted in in TypeScript trying to infer the schema type in two different ways (from the view, and from the root) when ended up getting bad inference (some code stopped compiling). This case is different since the node type is not the root type, so we don't get conflicting inference and it works as expected as is. Generally when possible we use the schema as the type parameter (if something other than just the node type is needed) since it tends to lead to the cleanest and most robust typing so I think this method should stay how it is: the above was just a workaround because it failed to work in that edge case.

Interestingly if we were using TypeScript 5.4, Noinfer on the second method parameter might have been able to fix the issue, but we can't do that yet.

Reply:
I see, that makes sense. Interesting, I hadn't heard about `noinfer` - cool!